module/src/utils/arrays.ts

```diff
@@ -27,6 +27,36 @@ export namespace Arrays {
     return Object.keys(dictionary).map<IArrayWithKey<T, Keys>>((key) => ({ key: key as Keys, items: dictionary[key] }));
   };
 
+  /**
+   * Re-indexes an array from a specific index point.
+   * - Does not mutate the passed array, returns a new one.
+   * @example If `["a", "b", "c", "d", "e"]` was passed with a `startFrom` index of `2`, the result would be `["c", "d", "e", "a", "b"]`
+   * @param array The array to clone and re-index.
+   * @param startFrom The current index of the new first item.
+   * @returns A new array, re-indexed with the item at the `startFrom` index now first.
+   */
+  export function reIndex<T>(array: T[], startFrom: number): T[] {
+    if (startFrom === 0) {
+      return array;
+    }
+    return [...[...array].splice(startFrom, array.length - startFrom), ...[...array].splice(0, startFrom)];
+  }
+
+  /**
+   * Creates a new array containing an item count that matches the passed `count`.
+   * - The items in the array are simply the index numbers.
+   * - Useful when you want to run a `map` x number of times but you don't have a specific array to loop.
```

Comment:
For that purpose I'd probably wrap up the map as well to remove a bit of boilerplate, that you could use like

```Arrays.repeat(index => <></>)```

Reply:
Happy to do this